docs/content/any/project/changelogs/NEXT.md

```diff
@@ -12,7 +12,29 @@ draft: true
 
 ### Core
 
+#### Breaking changes
+
+{{% callout "Warning" "orange" %}}
+  This release contains breaking changes. Be sure to follow migration steps
+  before upgrading.
+{{% /callout %}}
+
+##### Parenthesized variable dereferencing in specializer position is no longer permitted
+
+Previously, Oso would treat parenthesized specializers as variables. This made
+it possible to write rules with parameters that referenced each other, like so:
+
+```polar
+isa(x, y, x: (y));
```

Comment:
Is the current behavior equivalent to `isa(x, y, x) if x matches y;`? If so we could call that out here so anyone who's affected by this change will know how to fix it.

Reply:
I think it would be equivalent to `isa(x, y) if x matches y;` — the `isa/3` rule in the changelog is complemented by an `isa(x, y) if isa(x, y, x);` in @samscott89's original PR (#61)

Reply:
de38bb57

Reply:
So `isa` is actually a token in the lexer. So that example doesn't even work -- you get a parse error on unexpected token

Also, the rule doesn't even work any more. We'll try and unify x with y (since `y` isn't picked up as a tag, it's an external instance).

And the replacement suggestion doesn't work either. `isa(x, y) if x matches y` will actually do `x matches y{}` and you get `unregistered class y`.

And we also don't seem to support `x matches (y)` so that alternative doesn't work.

In conclusion... doing `rule(x, y, x: (y))` maybe _is_ pathological... it's a _really_ horrible way to write `rule(x, y, x)`. And there is nothing useful you can currently do with that syntax.

Reply:
Concretely. I'd remove the existing changelog entry, and replace it with a super minor entry like: "removed syntax for `f(x: (y))` which does not currently achieve anything".

Reply:
Oh, fun fact. The test cases with `x: (y)` in them output `x matches y` as constraints. But if you ground either side it'll ultimately end up attempting to do `x = y`.

Reply:
> Concretely. I'd remove the existing changelog entry, and replace it with a super minor entry like: "removed syntax for `f(x: (y))` which does not currently achieve anything".

256831d5